migrations/20201127170129-update-user-roles-in-entities-es.js

```diff
@@ -0,0 +1,86 @@
+module.exports = {
+  async up(db) {
+
+    global.migrationMsg = "Update user roles in entites elasticsearch";
+
+    if (!es) {
+      throw new Error("Elastic search connection not available.");
+    }
+
+    let entitiesIndex = process.env.ELASTICSEARCH_ENTITIES_INDEX;
+
+    if (!entitiesIndex || entitiesIndex == "") {
+      throw new Error("Invalid entities index");
+    }
+
+    const checkIfIndexExists = await es.indices.exists({ index: entitiesIndex });
+
+    if (checkIfIndexExists.statusCode !== 200) {
+      const createIndex = await es.indices.create({ index: entitiesIndex });
+      if (createIndex.statusCode != 200) {
+        throw new Error("Error while creating entities index.")
+      }
+    }
+
+    let users = await db.collection('userExtension').find({}).project({ _id: 1, roles: 1, userId: 1 }).toArray();
```

Comment:
@deepakr123 - Project only _id here then in chunks load the user from userExtension

Reply:
resolved